packages/block-editor/src/components/block-alignment-control/use-available-alignments.js

```diff
@@ -19,29 +19,43 @@ export default function useAvailableAlignments( controls = DEFAULT_CONTROLS ) {
 	if ( ! controls.includes( 'none' ) ) {
 		controls = [ 'none', ...controls ];
 	}
-	const {
-		wideControlsEnabled = false,
-		themeSupportsLayout,
-		isBlockBasedTheme,
-	} = useSelect( ( select ) => {
-		const { getSettings } = select( blockEditorStore );
-		const settings = getSettings();
-		return {
-			wideControlsEnabled: settings.alignWide,
-			themeSupportsLayout: settings.supportsLayout,
-			isBlockBasedTheme: settings.__unstableIsBlockBasedTheme,
-		};
-	}, [] );
+	const isNoneOnly = controls.length === 1 && controls[ 0 ] === 'none';
+
+	const [ wideControlsEnabled, themeSupportsLayout, isBlockBasedTheme ] =
+		useSelect(
+			( select ) => {
+				// If `isNoneOnly` is true, we'll be returning early because there is
+				// nothing to filter on an empty array. We won't need the info from
+				// the `useSelect` but we must call it anyway because Rules of Hooks.
+				// So the callback returns early to avoid block editor subscription.
+				if ( isNoneOnly ) {
+					return [ false, false, false ];
+				}
```

Comment:
Hopefully, the comment will mitigate this odd return value 😄

Reply:
Yes, it bothers me that it looks so weird 🙂 

On the other hand, this patch demonstrates a `useSelect` usage pattern that I think its quite neat and we could adopt it more widely: returning arrays instead of objects!

When writing:
```js
const { themeSupportsLayout } = useSelect( ( select ) => ( { themeSupportsLayout: ... } ) );
```
the `themeSupportsLayout` identifier is repeated twice, and constructing and destructuring the object is a bit unwieldy. Compare this to an array:
```js
const [ themeSupportsLayout ] = useSelect( ( select ) => ( [ ... ] ) );
```
That's much shorter and yet you don't lose any information. We're replacing the `themeSupportsLayout` with a `0` array index, basically. The field name doesn't matter because it's immediately destructured anyway.